includes/class-mailer.php

```diff
@@ -148,8 +149,12 @@ public static function new_follower( $notification ) {
 	 * @param int   $user_id  The id of the local blog-user.
 	 */
 	public static function direct_message( $activity, $user_id ) {
-		// Check if Activity is public or not.
-		if ( is_activity_public( $activity ) ) {
+		if (
+			is_activity_public( $activity ) ||
+			// Only accept messages that have the user in the "to" field.
+			empty( $activity['to'] ) ||
+			! in_array( ( new User( $user_id ) )->get_id(), (array) $activity['to'], true )
```

Comment:
how does this even work, the User class has no constructor that sets the ID 😳

I think this should be `Actors::get_by_id( $user_id )->get_id()` instead.

Reply:
Oops, thanks for catching it. Likely it worked because it used a 0-id in the test.